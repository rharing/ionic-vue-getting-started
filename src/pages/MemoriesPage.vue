<template>
 <base-layout page-title="All memories">
   <ion-list>
     <ion-item v-for="memory in memories"
               :router-link="`/memories/${memory.id}`"
               :key="memory.id"
     >
       <ion-thumbnail slot="start">
         <ion-img :src="memory.image" :alt="memory.title"/>
       </ion-thumbnail>
       <ion-label>
         {{ memory.title }}
       </ion-label>

     </ion-item>
   </ion-list>
 </base-layout>
</template>

<script setup lang="ts">
import {IonList, IonItem, IonImg, IonThumbnail,IonLabel } from '@ionic/vue';
import BaseLayout from "@/components/base/BaseLayout.vue";
const memories =
   [
  {
    id: 'm1',
    image:
        'https://upload.wikimedia.org/wikipedia/commons/thumb/7/76/Mighty_Mountains_with_Snow.jpg/640px-Mighty_Mountains_with_Snow.jpg',
    title: 'A trip into the mountains',
    description: 'It was a really nice trip!',
  },
  {
    id: 'm2',
    image:
        'https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/A_surfer_surfing_on_the_ocean_%28Unsplash%29.jpg/640px-A_surfer_surfing_on_the_ocean_%28Unsplash%29.jpg',
    title: 'Surfing the sea side',
    description: 'Feeling the cool breeze',
  },
  {
    id: 'm3',
    image:
        'https://upload.wikimedia.org/wikipedia/commons/thumb/a/a9/Indian_-_Food.jpg/640px-Indian_-_Food.jpg',
    title: 'Good eating',
    description: 'Really tasty!',
  },
];
</script>

<style scoped>

</style>
