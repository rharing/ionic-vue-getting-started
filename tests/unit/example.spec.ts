import { mount } from '@vue/test-utils'
import MemoriesPage from '@/pages/MemoriesPage.vue'
import { describe, expect, test } from 'vitest'

describe('MemoriesPage.vue', () => {
  test('renders MemoriesPage vue', () => {
    const wrapper = mount(MemoriesPage)
    expect(wrapper.text()).toMatchSnapshot();
  })
})
