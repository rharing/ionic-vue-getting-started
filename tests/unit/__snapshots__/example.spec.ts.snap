// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MemoriesPage.vue > renders MemoriesPage vue 1`] = `"All memoriesMaxManuelJulieGeorge"`;
